Valigator.Core/Core/Model.cs

```diff
@@ -88,16 +88,63 @@ private static Expression CreatePropertyDescriptor(Expression modelExpression, P
 
 			var modelExpression = Expression.Convert(modelParameter, typeof(TModel));
 
-			var validationErrors = typeof(TModel)
-				.GetProperties()
-				.Where(property => property.PropertyType.IsConstructedGenericType && property.PropertyType.GetGenericTypeDefinition() == typeof(Data<>))
+			var validationErrors = GetAllValigatorProperties(typeof(TModel))
 				.Select(property => VerifyProperty(modelExpression, property));
 
 			var arrayInitializer = Expression.NewArrayInit(typeof(ValidationError[]), validationErrors);
 
 			return Expression.Lambda<Func<TModel, ValidationError[][]>>(arrayInitializer, modelParameter).Compile();
 		}
 
+		private static PropertyInfo[] GetAllValigatorProperties(Type type)
+		{
+			var properties = 
+				GetBaseProperties(type)
+				.Concat(GetExplicitProperties(type))
+				.Where(property => property.PropertyType.IsConstructedGenericType && property.PropertyType.GetGenericTypeDefinition() == typeof(Data<>))
+				.ToArray();
+
+			var noGetters = properties.Where(x => x.GetMethod == null).ToArray();
+			var noSetters = properties.Where(x => x.SetMethod == null).ToArray();
+
+			if (noGetters.Any() || noSetters.Any())
+				throw new MissingGettersOrSettersException(noGetters, noSetters);
+
+			return properties;
+		}
+
+		private static IEnumerable<PropertyInfo> GetBaseProperties(Type type)
+		{
+			var currentLevelProperties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance);
```

Comment:
Since explicit properties are technically private, should we validate private properties? I'm still unsure.

Reply:
If we do, then we need to account for the fact that top level explicit properties will be returned by this method (if all we change is the bindingflags).